metrics/ci_integrations/lib/client/buildkite/buildkite_client.dart

```diff
@@ -96,6 +96,7 @@ class BuildkiteClient with LoggerMixin {
     return <String, String>{
       HttpHeaders.contentTypeHeader: ContentType.json.value,
       HttpHeaders.acceptHeader: ContentType.json.value,
+      HttpHeaders.userAgentHeader: null,
```

Comment:
What if we introduce the optional `headers` parameter for the client? There also be a private parameter `_headers` with values to apply to all API calls as follows:
```suggestion
      if (_headers != null) ..._headers,
```
In the constructor: `: _headers = headers;`
Thus, we can separate our requirements related to an empty `user-agent` from the low-level client implementation.